dolphinscheduler-ui/src/views/projects/workflow/definition/use-table.ts

```diff
@@ -257,6 +257,20 @@ export function useTable() {
         variables.page -= 1
       }
 
+      variables.checkedRowKeys = []
+      getTableData({
+        pageSize: variables.pageSize,
+        pageNo: variables.page,
+        searchVal: variables.searchVal
+      })
+    }).catch(() => {
+      if (
+          Math.ceil((variables.columns.length - variables.checkedRowKeys.length) / variables.pageSize) < variables.page
+          && variables.page > 1
+      ) {
+        variables.page -= 1;
+      }
+
       variables.checkedRowKeys = []
```

Comment:
There are still some problems:
- The current page is 1
- Why using columns length divide checkedRowKeys length
- It is better to set a common function to refresh table

Reply:
If delete some workflow in page 1 and all workflow of page 2,  and current page is 2,  after delete operation, current page will switch to page 1.  "columns length divide checkedRowKeys length" resolve this.

Reply:
Do we still have discussion here?

Reply:
Columns length means how many columns that the data table has, and it does not seem to be related to how many workflows that the data table has. It is same as the 'checkedRowkeys'.  So I didn't get your point. The page number need to be reduced by one, only in this case that it is the last page with one workflow.